[2025-02-18T16:41:44.539+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-02-18T16:41:44.571+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-02-18T16:41:44.588+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-02-18T16:41:44.588+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2025-02-18T16:41:44.615+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): transform_json_to_fact_table> on 2024-01-01 00:00:00+00:00
[2025-02-18T16:41:44.624+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=3786) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-02-18T16:41:44.627+0000] {standard_task_runner.py:72} INFO - Started process 3787 to run task
[2025-02-18T16:41:44.630+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pull_data_from_chess_api', 'transform_json_to_fact_table', 'scheduled__2024-01-01T00:00:00+00:00', '--job-id', '216', '--raw', '--subdir', 'DAGS_FOLDER/chess_dag.py', '--cfg-path', '/tmp/tmpnvq_hhvz']
[2025-02-18T16:41:44.632+0000] {standard_task_runner.py:105} INFO - Job 216: Subtask transform_json_to_fact_table
[2025-02-18T16:41:44.712+0000] {task_command.py:467} INFO - Running <TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [running]> on host 4ed02a11bf8e
[2025-02-18T16:41:44.868+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pull_data_from_chess_api' AIRFLOW_CTX_TASK_ID='transform_json_to_fact_table' AIRFLOW_CTX_EXECUTION_DATE='2024-01-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-01T00:00:00+00:00'
[2025-02-18T16:41:44.869+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-02-18T16:41:44.898+0000] {connection.py:531} ERROR - Unable to retrieve connection from secrets backend (EnvironmentVariablesBackend). Checking subsequent secrets backend.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 526, in get_connection_from_secrets
    conn = secrets_backend.get_connection(conn_id=conn_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/secrets/base_secrets.py", line 116, in get_connection
    return self.deserialize_connection(conn_id=conn_id, value=value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/secrets/base_secrets.py", line 70, in deserialize_connection
    return Connection(conn_id=conn_id, uri=value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __init__
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/state.py", line 481, in _initialize_instance
    with util.safe_reraise():
         ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/state.py", line 479, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 169, in __init__
    self._parse_from_uri(uri)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 257, in _parse_from_uri
    self.port = uri_parts.port
                ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as 'DefaultEndpointsProtocol=https;AccountName=rbchesssa;AccountKey=lG9W+utkpTNorE0rTLlfDMUZsTPleXk6pXv6NnfDTi'
[2025-02-18T16:41:44.916+0000] {base.py:84} INFO - Retrieving connection 'azure_chess_storage'
[2025-02-18T16:41:53.110+0000] {logging_mixin.py:190} INFO - /tmp/tmp430_tmhv
[2025-02-18T16:41:56.647+0000] {logging_mixin.py:190} INFO - Successfully Transformed Data and Loaded to Azure Storage as silver/fact-2024-01-games.csv
[2025-02-18T16:41:56.648+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-02-18T16:41:56.676+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-02-18T16:41:56.676+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=pull_data_from_chess_api, task_id=transform_json_to_fact_table, run_id=scheduled__2024-01-01T00:00:00+00:00, execution_date=20240101T000000, start_date=20250218T164144, end_date=20250218T164156
[2025-02-18T16:41:56.726+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-02-18T16:41:56.759+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-02-18T16:41:56.764+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-03-10T01:36:41.980+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-10T01:36:41.998+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-03-10T01:36:42.006+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [queued]>
[2025-03-10T01:36:42.006+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2025-03-10T01:36:42.019+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): transform_json_to_fact_table> on 2024-01-01 00:00:00+00:00
[2025-03-10T01:36:42.026+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=17053) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-03-10T01:36:42.028+0000] {standard_task_runner.py:72} INFO - Started process 17055 to run task
[2025-03-10T01:36:42.028+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'pull_data_from_chess_api', 'transform_json_to_fact_table', 'scheduled__2024-01-01T00:00:00+00:00', '--job-id', '524', '--raw', '--subdir', 'DAGS_FOLDER/collect_chess_data_dag.py', '--cfg-path', '/tmp/tmpm2oyvevi']
[2025-03-10T01:36:42.031+0000] {standard_task_runner.py:105} INFO - Job 524: Subtask transform_json_to_fact_table
[2025-03-10T01:36:42.079+0000] {task_command.py:467} INFO - Running <TaskInstance: pull_data_from_chess_api.transform_json_to_fact_table scheduled__2024-01-01T00:00:00+00:00 [running]> on host 70b3b9cfbba0
[2025-03-10T01:36:42.172+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='pull_data_from_chess_api' AIRFLOW_CTX_TASK_ID='transform_json_to_fact_table' AIRFLOW_CTX_EXECUTION_DATE='2024-01-01T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-01T00:00:00+00:00'
[2025-03-10T01:36:42.174+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-03-10T01:36:42.192+0000] {connection.py:531} ERROR - Unable to retrieve connection from secrets backend (EnvironmentVariablesBackend). Checking subsequent secrets backend.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 526, in get_connection_from_secrets
    conn = secrets_backend.get_connection(conn_id=conn_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/secrets/base_secrets.py", line 116, in get_connection
    return self.deserialize_connection(conn_id=conn_id, value=value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/secrets/base_secrets.py", line 70, in deserialize_connection
    return Connection(conn_id=conn_id, uri=value)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 4, in __init__
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/state.py", line 481, in _initialize_instance
    with util.safe_reraise():
         ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    compat.raise_(
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.12/site-packages/sqlalchemy/orm/state.py", line 479, in _initialize_instance
    return manager.original_init(*mixed[1:], **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 169, in __init__
    self._parse_from_uri(uri)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/connection.py", line 257, in _parse_from_uri
    self.port = uri_parts.port
                ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/urllib/parse.py", line 182, in port
    raise ValueError(f"Port could not be cast to integer value as {port!r}")
ValueError: Port could not be cast to integer value as 'DefaultEndpointsProtocol=https;AccountName=rbchesssa;AccountKey=lG9W+utkpTNorE0rTLlfDMUZsTPleXk6pXv6NnfDTi'
[2025-03-10T01:36:42.206+0000] {base.py:84} INFO - Retrieving connection 'azure_chess_storage'
[2025-03-10T01:36:44.974+0000] {logging_mixin.py:190} INFO - /tmp/tmpdg0culjk
[2025-03-10T01:36:45.097+0000] {logging_mixin.py:190} INFO - Previewing Dataframe
[2025-03-10T01:36:45.099+0000] {logging_mixin.py:190} INFO - game_url                        object
time_control                    object
rated                             bool
time_class                      object
rules                           object
white_rating                     int64
white_result                    object
black_rating                     int64
black_result                    object
pgn_event                       object
pgn_site                        object
game_date               datetime64[us]
pgn_white_user                  object
pgn_black_user                  object
pgn_result                      object
pgn_current_position            object
pgn_timezone                    object
pgn_eco                         object
pgn_eco_url                     object
start_time                      object
end_time                        object
end_game_date           datetime64[us]
pgn_raw                         object
pgn_trans                       object
dtype: object
[2025-03-10T01:36:45.122+0000] {logging_mixin.py:190} INFO - https://www.chess.com/game/live/97873251101	120+1	True	bullet	chess	472	timeout	513	win	Live Chess	Chess.com	2024-01-01 00:00:00	restlessbuddha	Rhythmbear1	0-1	2k1r3/1pprP1pp/1b1n1p2/5b2/1Q6/P1B2N2/2P2PPP/R3R1K1 w - -	UTC	A40	https://www.chess.com/openings/Englund-Gambit-Hartlaub-Charlick-Gambit	18:37:11	18:41:16	2024-01-01 00:00:00	d4 e5 dxe5 d6 exd6 Bxd6 Nc3 Nc6 Bd2 Bf5 e4 Bg6 Bb5 Qd7 Nf3 O-O-O O-O a6 Bxc6 Qxc6 e5 Bb4 a3 Ba5 b4 Bb6 Re1 f6 e6 Ne7 Qb1 Rhe8 b5 axb5 Nxb5 Qxb5 Qxb5 Nf5 Bc3 Nd6 Qb4 Bf5 e7 Rd7	1. d4 e5 2. dxe5 d6 3. exd6 Bxd6 4. Nc3 Nc6 5. Bd2 Bf5 6. e4 Bg6 7. Bb5 Qd7 8. Nf3 O-O-O 9. O-O a6 10. Bxc6 Qxc6 11. e5 Bb4 12. a3 Ba5 13. b4 Bb6 14. Re1 f6 15. e6 Ne7 16. Qb1 Rhe8 17. b5 axb5 18. Nxb5 Qxb5 19. Qxb5 Nf5 20. Bc3 Nd6 21. Qb4 Bf5 22. e7 Rd7
https://www.chess.com/game/live/97873345701	120+1	True	bullet	chess	507	win	505	timeout	Live Chess	Chess.com	2024-01-01 00:00:00	Pouyanpendar	Rhythmbear1	1-0	8/8/4k3/3p1p2/3P4/3N1K2/1p5P/8 b - -	UTC	C00	https://www.chess.com/openings/French-Defense-Knight-Variation-2...d5-3.exd5-exd5	18:41:30	18:47:38	2024-01-01 00:00:00	e4 e6 Nf3 d5 exd5 exd5 Nc3 c6 g3 Bd6 Bg2 Ne7 O-O O-O d3 Bg4 Bg5 f6 Bh4 Nd7 Qe2 Ne5 Qe3 Re8 Rae1 N7g6 d4 Nxf3+ Bxf3 Rxe3 Rxe3 Bxf3 Rxf3 Nxh4 gxh4 Qe8 Re3 Qg6+ Kh1 Qxc2 Re2 Qg6 Rg1 Qf7 f3 Re8 Reg2 g6 h5 Bb4 hxg6 hxg6 Rxg6+ Kf8 Rg8+ Ke7 R8g7 Qxg7 Rxg7+ Kf8 Rh7 Re1+ Kg2 Re2+ Kh3 Rxb2 Na4 Bc3 Nxc3 a5 f4 b5 f5 Rf2 Nd1 Rf3+ Kg2 Rxf5 Ne3 Rg5+ Kf3 Rg7 Rxg7 Kxg7 Nf5+ Kg6 Nd6 a4 Nb7 f5 Nc5 Kf6 Nd3 Ke6 Ne5 b4 Nxc6 b3 axb3 axb3 Nb4 b2 Nd3	1. e4 e6 2. Nf3 d5 3. exd5 exd5 4. Nc3 c6 5. g3 Bd6 6. Bg2 Ne7 7. O-O O-O 8. d3 Bg4 9. Bg5 f6 10. Bh4 Nd7 11. Qe2 Ne5 12. Qe3 Re8 13. Rae1 N7g6 14. d4 Nxf3+ 15. Bxf3 Rxe3 16. Rxe3 Bxf3 17. Rxf3 Nxh4 18. gxh4 Qe8 19. Re3 Qg6+ 20. Kh1 Qxc2 21. Re2 Qg6 22. Rg1 Qf7 23. f3 Re8 24. Reg2 g6 25. h5 Bb4 26. hxg6 hxg6 27. Rxg6+ Kf8 28. Rg8+ Ke7 29. R8g7 Qxg7 30. Rxg7+ Kf8 31. Rh7 Re1+ 32. Kg2 Re2+ 33. Kh3 Rxb2 34. Na4 Bc3 35. Nxc3 a5 36. f4 b5 37. f5 Rf2 38. Nd1 Rf3+ 39. Kg2 Rxf5 40. Ne3 Rg5+ 41. Kf3 Rg7 42. Rxg7 Kxg7 43. Nf5+ Kg6 44. Nd6 a4 45. Nb7 f5 46. Nc5 Kf6 47. Nd3 Ke6 48. Ne5 b4 49. Nxc6 b3 50. axb3 axb3 51. Nb4 b2 52. Nd3 
https://www.chess.com/game/live/97873876463	120+1	True	bullet	chess	513	win	517	checkmated	Live Chess	Chess.com	2024-01-01 00:00:00	Rhythmbear1	ludinho37	1-0	1k1Q4/ppp5/8/6p1/6B1/P3b3/1PP3PP/RK6 b - -	UTC	A06	https://www.chess.com/openings/Reti-Opening-Tennison-Gambit	18:48:17	18:52:25	2024-01-01 00:00:00	e4 d5 Nf3 dxe4 Ng5 Nf6 d3 Bf5 Bf4 Nc6 Be2 e5 Bg3 h6 dxe4 Qxd1+ Kxd1 O-O-O+ Bd3 hxg5 exf5 Bc5 Nd2 Ng4 Rf1 Rhe8 a3 e4 Be2 e3 fxe3 Rxe3 Bxg4 Ne5 Bxe5 Rxe5 f6+ Kb8 fxg7 Red5 Rxf7 Rxd2+ Kc1 Be3 Kb1 Rg8 Rf8+ Rxf8 gxf8=Q+ Rd8 Qxd8#	1. e4 d5 2. Nf3 dxe4 3. Ng5 Nf6 4. d3 Bf5 5. Bf4 Nc6 6. Be2 e5 7. Bg3 h6 8. dxe4 Qxd1+ 9. Kxd1 O-O-O+ 10. Bd3 hxg5 11. exf5 Bc5 12. Nd2 Ng4 13. Rf1 Rhe8 14. a3 e4 15. Be2 e3 16. fxe3 Rxe3 17. Bxg4 Ne5 18. Bxe5 Rxe5 19. f6+ Kb8 20. fxg7 Red5 21. Rxf7 Rxd2+ 22. Kc1 Be3 23. Kb1 Rg8 24. Rf8+ Rxf8 25. gxf8=Q+ Rd8 26. Qxd8# 
https://www.chess.com/game/live/97875744235	120+1	True	bullet	chess	502	resigned	382	win	Live Chess	Chess.com	2024-01-01 00:00:00	Rhythmbear1	Draknaard	0-1	r1bqkbnr/1pp2ppp/p1np4/1B2p3/4P3/5N2/PPPP1PPP/RNBQ1K1R w kq -	UTC	C62	https://www.chess.com/openings/Ruy-Lopez-Opening-Old-Steinitz-Defense	19:21:35	19:22:02	2024-01-01 00:00:00	e4 e5 Nf3 Nc6 Bb5 d6 Kf1 a6	1. e4 e5 2. Nf3 Nc6 3. Bb5 d6 4. Kf1 a6
https://www.chess.com/game/live/97875757239	120+1	True	bullet	chess	507	win	494	resigned	Live Chess	Chess.com	2024-01-01 00:00:00	Sportmen53	Rhythmbear1	1-0	rnb1k2r/ppp1bppp/4pn2/6N1/7Q/3B4/PP3PPP/RNB1K2R b KQkq -	UTC	C00	https://www.chess.com/openings/French-Defense	19:22:11	19:23:36	2024-01-01 00:00:00	e4 e6 Qh5 d5 Nf3 Nf6 Qh4 dxe4 Ng5 Be7 d3 exd3 cxd3 Qxd3 Bxd3	1. e4 e6 2. Qh5 d5 3. Nf3 Nf6 4. Qh4 dxe4 5. Ng5 Be7 6. d3 exd3 7. cxd3 Qxd3 8. Bxd3 
[2025-03-10T01:36:46.071+0000] {logging_mixin.py:190} INFO - Successfully Transformed Data and Loaded to Azure Storage as silver/fact-2024-01-games.parquet
[2025-03-10T01:36:46.072+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-03-10T01:36:46.083+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-03-10T01:36:46.084+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=pull_data_from_chess_api, task_id=transform_json_to_fact_table, run_id=scheduled__2024-01-01T00:00:00+00:00, execution_date=20240101T000000, start_date=20250310T013641, end_date=20250310T013646
[2025-03-10T01:36:46.141+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-03-10T01:36:46.170+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-03-10T01:36:46.173+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
